### ISSUE TITLE
Elasticsearch 7.16.1 Docker image fails to start

### ISSUE THREAD

**Issue by roizentner** [elastic/elasticsearch#81937]
**Elasticsearch version** (`bin/elasticsearch --version`):
Running Elasticsearch as a Statefulset in **local** Kubernetes cluster 
using elasticsearch:7.16.1.

**Description of the problem including expected versus actual behavior**:
Elasticsearch pods fail to start throwing the following error:
`the default discovery settings are unsuitable for production use; at least one of [discovery.seed_hosts, discovery.seed_providers, cluster.initial_master_nodes] must be configured`
Full log:
[elasticsearch-logs.txt](https://github.com/elastic/elasticsearch/files/7746339/elasticsearch-logs.txt)

**Steps to reproduce**:

 1. Apply Service.
 2. Apply Statefulset using elasticsearch:7.16.1 image.

YAML file used:
```
apiVersion: v1
kind: List
items:
- apiVersion: v1
  kind: Service
  metadata:
    name: elastic-service
    labels:
      app: elastic-service
  spec:
    selector:
      app: elasticsearch
    clusterIP: None
    ports:
    - port: 9200
      name: rest
    - port: 9300
      name: inter-node
- apiVersion: apps/v1
  kind: StatefulSet
  metadata:
    name: es-cluster
  spec:
    serviceName: elastic-service
    replicas: 3
    selector:
      matchLabels:
        app: elasticsearch
    template:
      metadata:
        labels:
          app: elasticsearch
      spec:
        serviceAccountName: elastic-serviceaccount
        affinity:
          podAntiAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
            - labelSelector:
                matchExpressions:
                - key: app
                  operator: In
                  values:
                  - elasticsearch
              topologyKey: "kubernetes.io/hostname"
        containers:
        - name: elasticsearch
          image: docker.elastic.co/elasticsearch/elasticsearch:7.16.1
          command:
          - /bin/sh
          - -cxe
          - |
            ulimit -l unlimited
            /usr/local/bin/docker-entrypoint.sh eswrapper
          securityContext:
            privileged: true
            capabilities:
              add:
              - IPC_LOCK
              - SYS_RESOURCE
          ports:
          - containerPort: 9200
            name: rest
            protocol: TCP
          - containerPort: 9300
            name: inter-node
            protocol: TCP
          volumeMounts:
          - name: esdata
            mountPath: /usr/share/elasticsearch/data
          env:
          - name: cluster.name
            value: my-elastic-cluster
          - name: node.name
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: cluster.initial_master_nodes
            value: "es-cluster-0,es-cluster-1,es-cluster-2"
          - name: discovery.seed_hosts
            value: "es-cluster-0.elastic-service,es-cluster-1.elastic-service,es-cluster-2.elastic-service"
          - name: xpack.security.enabled
            value: "false"
          - name: xpack.monitoring.enabled
            value: "false"
          - name: xpack.watcher.enabled
            value: "false"
          - name: xpack.graph.enabled
            value: "false"
          - name: xpack.ml.enabled
            value: "false"
          - name: node.ml
            value: "false"
          - name: "bootstrap.memory_lock"
            value: "true"
          - name: action.auto_create_index
            value: "false"
          - name: ES_JAVA_OPTS
            value: "-Xms2g -Xmx2g"
          resources:
              requests:
                cpu: 525m
                memory: 2500Mi
              limits:
                memory: 4Gi
        initContainers:
        - name: increase-vm-max-map
          image: busybox:1.31.1
          command: ["sysctl", "-w", "vm.max_map_count=262144"]
          securityContext:
            privileged: true
        - name: increase-fd-ulimit
          image: busybox:1.31.1
          command: ["sh", "-c", "ulimit -n 65536"]
          securityContext:
            privileged: true
    volumeClaimTemplates:
    - metadata:
        name: esdata
        labels:
          app: elasticsearch
      spec:
        accessModes: ["ReadWriteOnce"]
        storageClassName: local-storage
        resources:
          requests:
            storage: 20G
```

**Comment by roizentner** [comment]
I noticed the cluster name is *docker-cluster* and not as configured in my YAML file *my-elastic-cluster*. The same configuration worked in elasticsearch:7.6.2 and elasticsearch:7.14.2.

**Comment by pugnascotia** [comment]
It's quite possible that you've run into an issue that we fixed in https://github.com/elastic/elasticsearch/pull/81828 and released in v7.16.2, where running an entrypoint with `/bin/sh` causes issues. I recommend trying again with v7.16.2, or if there's some reason you can't, then try calling `/bin/bash` directly.

**Comment by roizentner** [comment]
@pugnascotia yep, that seems to have solved the issue. When I created the issue, v7.16.2 wasn't available yet on Docker Hub.
Thank you :)
